### ISSUE TITLE
v3.0.9 fails with System.InvalidOperationException : The Raven/DocumentsByEntityName index must exist in order to determine the document ID strategy.

### ISSUE THREAD

**Issue by johnsimons** [Particular/NServiceBus.RavenDB#229]
When attempting to upgrade https://github.com/Particular/ServiceControl/tree/Upgrade_To_Raven_v3 from v3.0.7 to v3.0.9 I get this error:
```
System.AggregateException : Test run failed due to one or more exception
  ----> NServiceBus.AcceptanceTesting.Support.ScenarioException : Endpoint Contexts.ManagementEndpoint failed to initialize
  ----> System.InvalidOperationException : The Raven/DocumentsByEntityName index must exist in order to determine the document ID strategy. This index is created by RavenDB automatically. Please check in Raven Studio to make sure it exists.
   at NServiceBus.AcceptanceTesting.Support.ScenarioRunner.Run(IList`1 runDescriptors, IList`1 behaviorDescriptors, IList`1 shoulds, Func`2 done, Int32 limitTestParallelismTo, Action`1 reports, Func`2 allowedExceptions) in C:\Projects\ServiceControl\src\ServiceControl.AcceptanceTesting\Support\ScenarioRunner.cs:line 87
   at NServiceBus.AcceptanceTesting.ScenarioWithContext`1.Run(RunSettings settings) in C:\Projects\ServiceControl\src\ServiceControl.AcceptanceTesting\ScenarioWithContext.cs:line 73
   at NServiceBus.AcceptanceTesting.ScenarioWithContext`1.NServiceBus.AcceptanceTesting.Support.IScenarioWithEndpointBehavior<TContext>.Run(Nullable`1 testExecutionTimeout) in C:\Projects\ServiceControl\src\ServiceControl.AcceptanceTesting\ScenarioWithContext.cs:line 112
   at ServiceBus.Management.AcceptanceTests.Audit.When_a_message_sent_from_third_party_endpoint_with_missing_metadata.Null_TimeSent_should_not_be_cast_to_DateTimeMin() in C:\Projects\ServiceControl\src\ServiceControl.AcceptanceTests\Audit\When_a_message_sent_from_third_party_endpoint_with_missing_metadata.cs:line 26
--ScenarioException
   at NServiceBus.AcceptanceTesting.Support.ScenarioRunner.InitializeRunners(RunDescriptor runDescriptor, IList`1 behaviorDescriptors) in C:\Projects\ServiceControl\src\ServiceControl.AcceptanceTesting\Support\ScenarioRunner.cs:line 348
   at NServiceBus.AcceptanceTesting.Support.ScenarioRunner.PerformTestRun(IList`1 behaviorDescriptors, IList`1 shoulds, RunDescriptor runDescriptor, Func`2 done, Func`2 allowedExceptions) in C:\Projects\ServiceControl\src\ServiceControl.AcceptanceTesting\Support\ScenarioRunner.cs:line 159
--InvalidOperationException
   at NServiceBus.RavenDB.Internal.DocumentIdConventions.GetTerms() in C:\Build\src\NServiceBus.RavenDB\Internal\DocumentIdConventions.cs:line 102
   at NServiceBus.RavenDB.Internal.DocumentIdConventions.MapTypeToCollectionName(Type type, CollectionData collectionData) in C:\Build\src\NServiceBus.RavenDB\Internal\DocumentIdConventions.cs:line 130
   at NServiceBus.RavenDB.Internal.DocumentIdConventions.Initialize() in C:\Build\src\NServiceBus.RavenDB\Internal\DocumentIdConventions.cs:line 69
   at NServiceBus.RavenDB.Internal.DocumentIdConventions.FindTypeTagName(Type type) in C:\Build\src\NServiceBus.RavenDB\Internal\DocumentIdConventions.cs:line 38
   at Raven.Client.Document.DocumentConvention.GetTypeTagName(Type type) in c:\Builds\RavenDB-Stable-3.0\Raven.Client.Lightweight\Document\DocumentConvention.cs:line 301
   at Raven.Client.Document.InMemoryDocumentSessionOperations.StoreInternal(Object entity, Etag etag, String id, Boolean forceConcurrencyCheck) in c:\Builds\RavenDB-Stable-3.0\Raven.Client.Lightweight\Document\InMemoryDocumentSessionOperations.cs:line 766
   at Raven.Client.Document.InMemoryDocumentSessionOperations.Store(Object entity) in c:\Builds\RavenDB-Stable-3.0\Raven.Client.Lightweight\Document\InMemoryDocumentSessionOperations.cs:line 698
   at NServiceBus.Unicast.Subscriptions.RavenDB.SubscriptionPersister.StoreNewSubscription(IDocumentSession session, String id, MessageType messageType) in C:\Build\src\NServiceBus.RavenDB\Subscriptions\SubscriptionPersister.cs:line 92
   at NServiceBus.Unicast.Subscriptions.RavenDB.SubscriptionPersister.<>c__DisplayClass5_0.<Subscribe>b__1(KeyValuePair`2 id) in C:\Build\src\NServiceBus.RavenDB\Subscriptions\SubscriptionPersister.cs:line 28
   at System.Linq.Enumerable.WhereSelectEnumerableIterator`2.MoveNext()
   at System.Linq.Enumerable.WhereEnumerableIterator`1.MoveNext()
   at System.Linq.Buffer`1..ctor(IEnumerable`1 source)
   at System.Linq.Enumerable.ToArray[TSource](IEnumerable`1 source)
   at NServiceBus.Unicast.Subscriptions.RavenDB.SubscriptionPersister.Subscribe(Address client, IEnumerable`1 messageTypes) in C:\Build\src\NServiceBus.RavenDB\Subscriptions\SubscriptionPersister.cs:line 27
   at ServiceControl.Infrastructure.SubscribeToOwnEvents.SubscribeForNonBrokers(Address address, IEnumerable`1 eventTypes) in C:\Projects\ServiceControl\src\ServiceControl\Infrastructure\SubscribeToOwnEvents.cs:line 39
   at ServiceControl.Infrastructure.SubscribeToOwnEvents.Run() in C:\Projects\ServiceControl\src\ServiceControl\Infrastructure\SubscribeToOwnEvents.cs:line 25
   at Particular.ServiceControl.Bootstrapper.ConfigureNServiceBus() in C:\Projects\ServiceControl\src\ServiceControl\Bootstrapper.cs:line 117
   at Particular.ServiceControl.Bootstrapper.Start() in C:\Projects\ServiceControl\src\ServiceControl\Bootstrapper.cs:line 162
   at ServiceBus.Management.AcceptanceTests.Contexts.ManagementEndpointSetup.<>c__DisplayClass0_0.<GetConfiguration>b__2() in C:\Projects\ServiceControl\src\ServiceControl.AcceptanceTests\Contexts\ManagementEndpointSetup.cs:line 49
   at NServiceBus.AcceptanceTesting.Support.EndpointRunner.Initialize(RunDescriptor run, EndpointBehavior endpointBehavior, IDictionary`2 routingTable, String endpointName) in C:\Projects\ServiceControl\src\ServiceControl.AcceptanceTesting\Support\EndpointRunner.cs:line 61
```

To repro it, checkout https://github.com/Particular/ServiceControl/tree/Upgrade_To_Raven_v3 and run the ATs, these should work.
Then update NServiceBus.RavenDB package to v3.0.9
The error above should be visible in some tests:
![image](https://cloud.githubusercontent.com/assets/122651/15065507/c6b27342-13a3-11e6-8dff-c8f7565919f9.png)

**Comment by mauroservienti** [comment]
> The Raven/DocumentsByEntityName index must exist in order to determine the document ID strategy. This index is created by RavenDB automatically. Please check in Raven Studio to make sure it exists.

However not on embedded :cry:

@DavidBoike the package at startup needs to ensure that the index exists, it is enough to call the `IndexCreation.CreateIndex( new DocumentsByEntityName() )` (or something like that, on the phone now), if the index is already there nothing happens, otherwise it will be created.

**Comment by mauroservienti** [comment]
thanks @johnsimons :smile: good catch. Would be interesting to understand why the samples smoke tests didn't catch this since, IIRC, they run on Embedded (at least some of them).

**Comment by johnsimons** [comment]
@DavidBoike how is this going ?

**Comment by DavidBoike** [comment]
@johnsimons Suffering from not enough maintainers. Want to take a look at my latest rundown in the Raven channel in Slack and review those PRs?

**Comment by DavidBoike** [comment]
Fixed by 3.0.10, and nominally only affects ServiceControl (embedded RavenDB databases don't get the index initialized on every connection) anyway.
